pkg/controllers/node/node_controller.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019 Red Hat and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"context"
+	"github.com/nimbess/stargazer/pkg/config"
+	"github.com/nimbess/stargazer/pkg/controller"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"sync"
+)
+
+type NodeController struct {
```

Comment:
should this be composed from a generic Controller struct? That way when you add other controllers they can all just inherit from that type?

Reply:
Some of it might be common, like context and config. There does need to be a distinct structure though since each controller is different an has some unique data. When I add more meat to the node controller and add the unp controller I will look at refactoring common parts.